src/DataTable/BulkActions.jsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { MoreVert } from '../../icons';
+import {
+  Button, Dropdown, useWindowSize, Icon, breakpoints,
+} from '..';
+
+export const DROPDOWN_BUTTON_TEXT = 'More actions';
+export const SMALL_SCREEN_DROPDOWN_BUTTON_TEXT = 'Actions';
+
+// eslint-disable-next-line react/prop-types
+const CustomToggle = React.forwardRef(({ children, onClick }, ref) => (
+  <Button
+    href=""
+    ref={ref}
+    variant="tertiary"
+    onClick={(e) => {
+      e.preventDefault();
+      onClick(e);
+    }}
+  >
+    {children}
+  </Button>
+));
+
+const BulkActions = ({
+  actions, selectedRows, className, ...rest
+}) => {
+  const { width } = useWindowSize();
+  const [visibleActions, dropdownActions] = useMemo(() => {
+    if (width < breakpoints.small.minWidth) {
+      // On a small screen, all actions will be in the overflow menu
+      return [[], [...actions]];
+    }
+    // The first two actions will be displayed as buttons, the rest will go in an overflow menu
+    const firstTwoActions = [...actions].splice(0, 2);
+    const extraActions = [...actions].slice(2);
```

Comment:
Note for myself mostly that this scenario may be a good reason to pull this hook into Paragon https://github.com/edx/frontend-app-learning/blob/master/src/generic/tabs/useIndexOfLastVisibleChild.js

Reply:
Oh, yeah, that looks super useful